src/pybind/mgr/cephadm/module.py

```diff
@@ -1792,17 +1792,42 @@ def remove_daemons(self, names):
         return self._remove_daemons(args)
 
     @handle_orch_error
-    def remove_service(self, service_name: str) -> str:
+    def remove_service(self, service_name: str, force: bool = False) -> str:
         self.log.info('Remove service %s' % service_name)
         self._trigger_preview_refresh(service_name=service_name)
+        osd_removal_completion = None
+
         if service_name in self.spec_store:
             if self.spec_store[service_name].spec.service_type in ('mon', 'mgr'):
                 return f'Unable to remove {service_name} service.\n' \
                        f'Note, you might want to mark the {service_name} service as "unmanaged"'
+
+            # OSD services removal
+            if self.spec_store[service_name].spec.service_type == 'osd':
```

Comment:
```suggestion
            if service_name.startswith('osd.'):
```

we should probably provide a `def service_type_by_name(service_name: str): .. ` helper somewhere.

Reply:
I think it is more easy and more safe to check the "service_type" attribute instead of trying to guess the service type using the service name.